src/ui/zcl_abapgit_gui_chunk_lib.clas.abap

```diff
@@ -153,6 +158,100 @@ CLASS zcl_abapgit_gui_chunk_lib IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD render_error_message_box.
+
+    CONSTANTS: lc_regex_msgid_and_msgno TYPE string VALUE `(.{5})` ##NO_TEXT.
+
+    DATA:
+      lv_error_text      TYPE string,
+      lv_longtext        TYPE string,
+      lv_msgid_and_msgno TYPE string,
+      lv_program_name    TYPE syrepid.
+
+
+    CREATE OBJECT ro_html.
+
+    lv_error_text = ix_error->get_text( ).
+    lv_longtext = ix_error->get_longtext( abap_true ).
+
+    REPLACE ALL OCCURRENCES OF cl_abap_char_utilities=>newline
+            IN lv_longtext
+            WITH '<br>'.
+
+    FIND FIRST OCCURRENCE OF REGEX lc_regex_msgid_and_msgno
+         IN lv_longtext
+         SUBMATCHES lv_msgid_and_msgno.
+
+    IF sy-subrc = 0.
+      REPLACE FIRST OCCURRENCE OF REGEX lc_regex_msgid_and_msgno
+              IN lv_longtext
+              WITH ``.
+    ENDIF.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-cause }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-system_response }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-what_to_do }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-sys_admin }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    ro_html->add( |<div id="message" class="message-panel">| ).
+    ro_html->add( |{ lv_error_text }| ).
+    ro_html->add( |<div class="float-right">| ).
+
+    ro_html->add_a(
+        iv_txt   = `&#x274c;`
+        iv_act   = `toggleDisplay('message')`
+        iv_class = `close-btn`
+        iv_typ   = zif_abapgit_html=>c_action_type-onclick ).
+
+    ro_html->add( |</div>| ).
+
+    ro_html->add( |<div class="float-right message-panel-commands">| ).
+
+    IF lv_msgid_and_msgno IS NOT INITIAL.
+      ro_html->add_a(
+          iv_txt = lv_msgid_and_msgno
+          iv_typ = zif_abapgit_html=>c_action_type-sapevent
+          iv_act = zif_abapgit_definitions=>c_action-goto_message
+          iv_id  = `a_goto_message` ).
+    ENDIF.
+
+    ix_error->get_source_position(
+      IMPORTING
+        program_name = lv_program_name ).
+
+    ro_html->add_a(
+        iv_txt   = `Goto source`
+        iv_act   = zif_abapgit_definitions=>c_action-goto_source
+        iv_typ   = zif_abapgit_html=>c_action_type-sapevent
+        iv_title = |{ lv_program_name }|
+        iv_id    = `a_goto_source` ).
```

Comment:
Currently it looks like e.g. `"ZCL_ABAPGIT_TAG_POPUPS========CP"`, I'd propose to remove suffix and add method name - would be more descriptive. (maybe a separate PR ...)

Here is an example of a comparable feature from one of my devs

```abap
method internal_error.
  data:
        l_prog      type string,
        l_meth      type string,
        sys_call    type sys_calls,
        t_sys_stack type sys_callst.

  call function 'SYSTEM_CALLSTACK'
    exporting
      max_level    = 2
    importing
      et_callstack = t_sys_stack.

  read table t_sys_stack into sys_call index 2.
  l_prog     = sys_call-progname.
  l_meth     = sys_call-eventname. " <<<< METHOD

  if sys_call-eventtype = 'METH'.
    data oref type ref to if_oo_class_incl_naming.
    data oclif type ref to if_oo_clif_incl_naming.

    call method cl_oo_include_naming=>get_instance_by_include
      exporting
        progname       = sys_call-progname
      receiving
        cifref         = oclif
      exceptions
        no_objecttype  = 1
        internal_error = 2.
    if sy-subrc is initial.
      oref ?= oclif.
      l_prog = oref->clskey-clsname. " <<< clean class name
    endif.
  endif.

  raise exception type xxxxxxxxx
    exporting
      textid    = m001_internal_error
      str_attr1 = l_prog
      str_attr2 = l_meth
      str_attr3 = text
      chr2      = code.

endmethod.
```